### DISCUSSION TITLE
TraceId, SpanId are changed while using opentelemetry-javaagent for Distributed logging and tracing in a java microservice based architecture.

### DISCUSSION THREAD

**Discussion by manukumar2297** [open-telemetry/opentelemetry-java-instrumentation#11452] [Q&A]
Hi all,

I'm using opentelemetry-javaagent (v2.3.0) for distributed logging of a micro service based architecture java app to decorate the logs with TraceId and spanId. For an event in my app the control goes through different Microservices to get the end result.
By running the Opentelemetry-javaagent with the required environment variables, the logs are decorated with the TraceId and spanID across different microservices, (The logs associated with an event should have the sameTraceId across different services, so that we'll be able to track the logs for that event using that traceId)
But at one point when the control goes from Microservice3 to Microservice4 and comes back to Microservice3, the new TraceId is generated and going forward the new TraceId is considered and printed in the logs. Is it like the context is not getting passed from Microservice4 to Microservice3 or are we missing anything else ? Any Inputs will be helpful here, Thanks in Advance....

**Comment by laurit** [comment]
Perhaps one of these services uses a framework that we don't have an instrumentation with so context propagation does not happen? You could examine the request on the receiving side and see whether context propagation headers are present.

**Reply by manukumar2297** [reply]
We are using Kafka event driven microservices that interact with a central database.
Based on the event in Microservice1 the Kafka Listener in the other microservice executes.
